packages/scaffold-ui/src/partials/w3m-list-wallet/index.ts

```diff
@@ -62,16 +61,19 @@ export class W3mListWallet extends LitElement {
     super.updated(changedProperties)
 
     // Reset impression tracking when wallet changes
-    if (changedProperties.has('name') || changedProperties.has('imageSrc')) {
+    if (
+      changedProperties.has('name') ||
+      changedProperties.has('imageSrc') ||
+      changedProperties.has('walletRank')
```

Comment:
This means that 
1) the w3m-list-wallet doesn't have walletRank property
2) it get's updated walletRank and triggers impression event right?

So the getting the walletRank happens async right?

Reply:
yea, given we need to fetch the ranks on connector-list render this will get updated, so I want to trigger the impression when wallet rank has changed and is present.